integration_tests/src/main/python/string_test.py

```diff
@@ -823,24 +823,19 @@ def test_format_number_supported(data_gen):
     )
 
 float_format_number_conf = {'spark.rapids.sql.formatNumberFloat.enabled': 'true'}
-format_number_float_gens = [DoubleGen(min_exp=-300, max_exp=15)]
-
-@pytest.mark.parametrize('data_gen', format_number_float_gens, ids=idfn)
-def test_format_number_float_limited(data_gen):
+format_number_float_gens = [(DoubleGen(), 0.05), (FloatGen(), 0.5), 
+                            (SetValuesGen(FloatType(), [float('nan'), float('inf'), float('-inf'), 0.0, -0.0]), 1.0),
+                            (SetValuesGen(DoubleType(), [float('nan'), float('inf'), float('-inf'), 0.0, -0.0]), 1.0)]
+# The actual error rate is 2% for double and 42% for float
+# set threshold to 5% and 50% to avoid bad luck
+
+@pytest.mark.parametrize('data_gen,max_err', format_number_float_gens, ids=idfn)
+def test_format_number_float_limited(data_gen, max_err):
     gen = data_gen
-    assert_gpu_and_cpu_are_equal_collect(
-        lambda spark: unary_op_df(spark, gen).selectExpr(
-            'format_number(a, 5)'),
-        conf = float_format_number_conf
-    )
-
-# format_number for float/double is disabled by default due to compatibility issue
-# GPU will generate result with less precision than CPU
-@allow_non_gpu('ProjectExec')
-@pytest.mark.parametrize('data_gen', [float_gen, double_gen], ids=idfn)
-def test_format_number_float_fallback(data_gen):
-    assert_gpu_fallback_collect(
-        lambda spark: unary_op_df(spark, data_gen).selectExpr(
-            'format_number(a, 5)'),
-        'FormatNumber'
-    )
+    cpu = with_cpu_session(lambda spark: unary_op_df(spark, gen).selectExpr('*',
+            'format_number(a, 5)').collect(), conf = float_format_number_conf)
+    gpu = with_gpu_session(lambda spark: unary_op_df(spark, gen).selectExpr('*',
+            'format_number(a, 5)').collect(), conf = float_format_number_conf)
+    mismatched = sum(x[0] != x[1] for x in zip(cpu, gpu))
```

Comment:
I preferred the version that checked that when we parsed them back to a float the numbers were within the error bounds instead of saying that we cannot be wrong more than some set percentage.

Reply:
done.